btk/obs_conditions.py

```diff
@@ -114,6 +114,34 @@ def get_psf_sky(self, psf_stamp_size):
         return psf_image, mean_sky_level
 
 
+class CosmosCutout(Cutout):
+    @staticmethod
+    def psf_function(r):
+        # usually r = 0.3 * pix
+        return galsim.Moffat(2, r)
+
+    def get_psf(self, psf_stamp_size):
+        assert psf_stamp_size % 2 == 1
+        psf_int = self.psf_function(self.psf_args).withFlux(1.0)
+
+        # Draw PSF
+        psf = psf_int.drawImage(
+            nx=self.psf_size,
+            ny=self.psf_size,
+            method="real_space",
+            use_true_center=True,
+            scale=self.pix,
+        ).array
+
+        # Make sure PSF vanishes on the edges of a patch that
+        # has the shape of the initial npsf
+        psf = psf - psf[0, int(self.psf_size / 2)] * 2
+        psf[psf < 0] = 0
+        psf = psf / np.sum(psf)
+
```

Comment:
```suggestion
## Make sure PSF vanishes on the edges of a patch that has the shape of the initial npsf
            
            
            psf_obj = galsim.InterpolatedImage(galsim.Image(psf), scale=self.pix).withFlux(1.)

            ## Interpolate the new 0-ed psf
            psfs_obj.append(psf_obj)

            ## Re-draw it (with the correct fulx)
            psfs.append(psf_obj.drawImage(nx=npsf,
                                          ny=npsf,
                                          method = 'no_pixel',
                                          use_true_center = True,
                                          scale = self.pix).array)
```

Reply:
I think the draw has to be with no pixel here, because pixel integration already happened.